Update baremetalinstance version

alexandraavendano opened the pull request:
Introducing a new 2.0.0 version of the bare metal infrastructure extension. This update includes a powerful new force power management command specifically designed for SuperDome Flex Servers.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az extension add --name baremetal-infrastructure` should install the new 2.0.0 version instead of old 1.0.0 version.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [X] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).


### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

yonzhan commented:
<!-- add-pr-comment:add-pr-comment -->

Thank you for your contribution! We will review the pull request and get back to you soon.

alexandraavendano commented:
I've noticed that two pipelines are failing with an error indicating the wheel version should be "generator": "bdist_wheel (0.30.0)",, but I haven't made any changes to that section of the index.json file. Could you please provide some guidance on how to resolve this error? Thank you!

kairu-ms commented:
Hi @bebound, do you have any context of the failed CI task?

bebound commented:
I can't find `metadata.json` in wheel package. I guess you used a newer `wheel` to build the package instead of 0.30.0.

PS: Did you intentionally remove 1.0 from the index?

alexandraavendano commented:
Hi @bebound ! Thanks for your advice. I've updated the wheel version in our release repo. We're keeping versions 1.0.0 and 2.0.0, so I've added them back to our list. Really appreciate your quick and helpful response! Now the pipelines are healthy.
